### PR TITLE
Virtiofs, configvolumes: Add how to share configvolumes with virtiofs.

### PR DESCRIPTION
**Pull request by jcanocan**
This PR adds details about how to use the enhancement introduced in https://github.com/kubevirt/kubevirt/pull/9432 to share ConfigMaps, Secrets, DownwardAPI and ServiceAccounts through Virtiofs.

### CODE REVIEW

File: docs/virtual_machines/disks_and_volumes.md
Comment (lyarwood) [inline docs/virtual_machines/disks_and_volumes.md:697]: If disk is the default behaviour you shouldn't need to add a reference here, the VMI mutation webhook should add this for us during submission.

File: docs/virtual_machines/disks_and_volumes.md
Comment (lyarwood) [inline docs/virtual_machines/disks_and_volumes.md:826]: As above, might not be required if this is the default behaviour.

File: docs/virtual_machines/disks_and_volumes.md
Comment (lyarwood) [inline docs/virtual_machines/disks_and_volumes.md:943]: In this example you've used `virtio` as the disk bus so adding a disk is required without preferences but assuming the bus isn't important you can again drop this from the example to illustrate the default behaviour.

**Review by lyarwood — COMMENTED** [review]
Looks good overall, quick note about the default behaviour here, we might be able to drop the actual disk definition(s) to make it clear to the reader.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md:942]: I've dropped the `virtio` bus from the example. Do you think that I should drop the `bus`  from the other examples as well?

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Dropped the `disk: {}`  I still need the disk section to set the `serial`.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: See above.

**Comment by jcanocan** [comment]
/hold 
We are waiting to have this PR https://github.com/kubevirt/kubevirt/pull/9680 merged. It will remove the need to have the feature gate. Therefore, some notes will be dropped.

**Comment by jcanocan** [comment]
@cwilkers Could you please take a look when you have some availability?

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [reply docs/virtual_machines/disks_and_volumes.md:727]: Also, in that Disks section, it would be important to provide links to virtiofs documentation, explain the benefits versus disk mode, and include the live-migration note both at the top, and in each "filesystem" subsection.

**Review by cwilkers — CHANGES_REQUESTED** [review]
This is an excellent new set of features, they need more mention in the top of the document, as noted inline.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Done.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md:680]: Done.

**Review by cwilkers — CHANGES_REQUESTED** [review]
Thanks! I have a couple edit suggestions on the new section; the section at the top is looking good.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Done

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Done

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Done

**Comment by cwilkers** [comment]
/lgtm
I'm ready to approve this, do you want an extra review from somebody else first?

**Review by cwilkers — APPROVED** [review]

File: docs/virtual_machines/disks_and_volumes.md
Comment (germag) [inline docs/virtual_machines/disks_and_volumes.md:137]: This should be: "A `filesystem` device"

 it's a _device_ not a _disk_. Also, I think `filesystems`  it should be in singular

**Comment by germag** [comment]
Besides that minor fix, it's lgtm too. Thanks

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Addresed.

**Comment by jcanocan** [comment]
@cwilkers could you please add the lgtm tag again?

**Comment by alicefr** [comment]
@jcanocan Can we add a brief description of how to identify the mount name that needs to be used in cloud-init with virtiofs. From the examples, it is quite clear, however, I think a generic description will help users.

**Comment by jcanocan** [comment]
> @jcanocan Can we add a brief description of how to identify the mount name that needs to be used in cloud-init with virtiofs. From the examples, it is quite clear, however, I think a generic description will help users.

I've added an explanation as well as a generic example.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Addressed.

**Comment by jcanocan** [comment]
> IMO, we should also mention that for old OSes/kernels, virtiofs filesystem isn't supported and this new options don't work. Maybe also we should add some hints on how to check if the OS support virtiofs as a filesystem.

I've added a note pointing out this. Please check if everything is accurate.

**Comment by jcanocan** [comment]
> Thanks, @jcanocan looks good to me!

Many thanks for your reviews :)

**Comment by jcanocan** [comment]
@cwilkers I think this PR is good to go.

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:146]:
```suggestion
Additionally, `filesystem` devices must be mounted inside the VM.
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:727]:
```suggestion
sections and both have some advantages and disadvantages, i.e. `disk` does not support dynamic change propagation and
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:728]:
```suggestion
`filesystem` does not support live migration.
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:729]:
```suggestion
Therefore, depending on the use-case, one or the other may be more suitable.
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:800]:
```suggestion
`filesystem` allows you to dynamically propagate changes on `configMaps` to VMIs (i.e. the VM does not need to be rebooted).
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:856]:
```suggestion
sections and both have some advantages and disadvantages, e.g. `disk` does not support dynamic change propagation and
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:857]:
```suggestion
`filesystem` does not support live migration.
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:858]:
```suggestion
Therefore, depending on the use-case, one or the other may be more suitable.
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:928]:
```suggestion
`filesystem` allows you to dynamically propagate changes on `secrets` to VMIs (i.e. the VM does not need to be rebooted).
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:985]:
```suggestion
Therefore, depending on the use-case, one or the other may be more suitable.
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:1030]:
```suggestion
By using filesystem, `serviceAccounts` are shared through `virtiofs`. In contrast with using disk for sharing `serviceAccounts`,
```

File: docs/virtual_machines/disks_and_volumes.md
Comment (cwilkers) [inline docs/virtual_machines/disks_and_volumes.md:1031]:
```suggestion
`filesystem` allows you to dynamically propagate changes on `serviceAccounts` to VMIs (i.e. the VM does not need to be rebooted).
```

**Review by cwilkers — CHANGES_REQUESTED** [review]
There are a few more small issues with singular/plural and i.e. vs e.g. (I think of i.e. as "in other words", and e.g. as "example given" to help me remember)

Otherwise, great work, I'll approve and lgtm once you resolve these bits of feedback.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Addressed this and the rest of occurrences.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Addressed.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Done.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Addressed.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Done.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Fixed.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Done.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Addressed.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Done.

File: docs/virtual_machines/disks_and_volumes.md
Comment (jcanocan) [reply docs/virtual_machines/disks_and_volumes.md]: Done.

**Comment by cwilkers** [comment]
Thanks for your patience with the review process!

/approve
/lgtm

**Pull request merged**
